pkg/infra/iac2/templates_compiler.go

```diff
@@ -565,7 +572,7 @@ func (tp templatesProvider) getTemplateForType(typeName string) (ResourceCreatio
 	return template, nil
 }
 
-func (tp templatesProvider) getNestedTemplate(templatePath string) (*template.Template, error) {
+func (tp templatesProvider) getNestedTemplate(templatePath string, funcs template.FuncMap) (*template.Template, error) {
```

Comment:
I don't think this method of supplying functions is going to work with caching of parsed templates since the function and its context are ignored after the first time we get the template. We probably need a single up-front `parseVal` function declaration that we supply with a reference to some context holder that we update prior to each attempt to parse a nested template value.

Reply:
alright, i had just copied what you had done previously, but i can rework it. Youre thinking of something like 

```
func(val reflect.Value) (string, error) {
					return tc.resolveStructInput(nil, val, useDoubleQuotedStrings, "", appliedOutputs)
				},
			}
```

needs to be defined staticly somewhere and just referenced in the `	tmpl, err := template.New(templatePath).Funcs(funcs).Parse(string(contents))
`
line?

Reply:
partially, but you'd also need to do something like this:

```go
type (
  TemplatesCompiler struct {
    ...
    Ctx *NestedCtx
  }

  NestedCtx struct {
    useDoubleQuotes bool
    appliedOutputs []AppliedOutput
    rootVal reflect.Value
  }
)

...

// parseVal
func(val reflect.Value) (string, error) {
  return tc.resolveStructInput(tc.Ctx.rootVal, val, tc.Ctx.useDoubleQuotedStrings, "", tc.Ctx.appliedOutputs)
}
```

Then make sure to update `tc.Ctx` before each attempt to parse a value in a nested template.